### PR TITLE
Consolidate Load and Save Functions 

### PR DESCRIPTION
**Pull request by Mike-3**
convert string to bytes
only 1 export file routine

### CODE REVIEW

**Comment by S-Man42** [comment]
Interesting is still that the load routine is still untouched. So, the back-operation still does not reconvert the loaded bytes correctly into a String. convertBytesToString is never called, isnt it?

**Comment by Mike-3** [comment]
We create only GCW files with byte data. So it has to be called individually every time it is converted to a string.

**Comment by S-Man42** [comment]
this makes sense

File: lib/utils/ui_dependent_utils/file_widget_utils.dart [hunk 1]
```diff
@@ -11,15 +11,23 @@ import 'package:gc_wizard/utils/file_utils/file_utils.dart';
...
+}
+
+Future<bool> saveStringToFile(BuildContext context, String data, String fileName) async {
+  return _saveDataToFile(context, convertStringToBytes(data), fileName, true);
+}
+
+Future<bool> _saveDataToFile(BuildContext context, Uint8List data, String fileName, bool byteData) async {
   if (kIsWeb) {
-    var blob = html.Blob([data], 'image/png');
+    var blob = html.Blob([data], byteData ? 'image/png' : 'text/plain', byteData ? null : 'native');
```
Comment (S-Man42) [inline lib/utils/ui_dependent_utils/file_widget_utils.dart:23]: Are you quite sure that this really works? So you say: byteData IS ALWAYS an PNG image. But we do save any other file type as well (hidden data, wig analyzer...). Do I miss something?

**Comment by Mike-3** [comment]
I have seen too.
That's how it was before and we never had any problems with it, right?
I also don't know what should change when saving files if we specify other types there.

**Comment by S-Man42** [comment]
The the MIME Type has to be taken from the GCWFile File Type.

File: lib/utils/ui_dependent_utils/file_widget_utils.dart [hunk 2]
```diff
@@ -20,7 +20,11 @@ Future<bool> saveStringToFile(BuildContext context, String data, String fileName
 
 Future<bool> _saveDataToFile(BuildContext context, Uint8List data, String fileName, bool byteData) async {
   if (kIsWeb) {
-    var blob = html.Blob([data], byteData ? 'image/png' : 'text/plain', byteData ? null : 'native');
+    var fileType = fileTypeByFilename(fileName);
+    String? mimeType;
```
Comment (S-Man42) [inline lib/utils/ui_dependent_utils/file_widget_utils.dart:24]: What happens on fileType == null?

**Comment by Mike-3** [comment]
No problem. The parameter is optional.

**Comment by S-Man42** [comment]
then saveStringToFile should add text/plain?

**Comment by Mike-3** [comment]
Si

**Comment by S-Man42** [comment]
Ok :D Do you add this? :) :)

**Pull request merged**